### DISCUSSION TITLE
The ocr_predictor utilizes only 1 CPU core.

### DISCUSSION THREAD

**Discussion by sergei-voronkov** [mindee/doctr#1703] [General]
I have a set of images from real documents.
While I try to recognize text with default ocr_predictor setting + torch, it takes about 10 seconds per page.
I notice the CPU load it too low in the same time. It seems the model uses only 1 CPU core for the recognition.
Is there a way to utilize all my core to speed up the recognition.

My configuration:
Windows 10
Python 3.12.4
Pytorch
No GPU

Thanks!

**Comment by felixdittrich92** [comment]
Hi @sergei-voronkov :wave:,

Looks like an issue with the PyTorch installation on your machine, this does not come from docTR itself.
Could you try to force it with `torch.set_num_threads(num_cores)` ?

Otherwsie you can also try: https://github.com/felixdittrich92/OnnxTR which is optimized for inference, especially on CPU / and or limited hardware.

Best regards,
Felix

**Reply by felixdittrich92** [reply]
ref.: https://github.com/pytorch/pytorch/issues/9873

**Reply by sergei-voronkov** [reply]
Hi @felixdittrich92 !
Thanks for your advices!

First, I'll definitely try the OnnxTR.

About the `torch.set_num_threads(num_cores)`.  I've tried to force multithreading with `torch.set_num_threads(num_cores)`, but after the call, the `ocr_predictor` predicts nothing:

## Image
![image](https://github.com/user-attachments/assets/4d164d36-4203-4a4c-8656-f0c3d2a45f68)

## num_thread == 1

![image](https://github.com/user-attachments/assets/9af18a18-11da-48c9-a7a5-e2570447b102)

`prediction.reader == 'Hello, World!\nMy name is Sergey.\nMy position is Senior Software Developer at STO Solutions'`

## num_threads == 2

`prediction.reader == ''`

**Reply by felixT2K** [reply]
Have you installed pytorch with pip or conda ? :)

**Reply by sergei-voronkov** [reply]
I've installed it with pip in a dedicated conda environment.

**Reply by felixT2K** [reply]
Could you try to install with conda instead ?

**Reply by sergei-voronkov** [reply]
The conda-base torch fixes the problem! Thanks a lot!

**Reply by felixdittrich92** [reply]
Nice 😊

**Reply by sergei-voronkov** [reply]
Could you explain the reason? Wrong torch version?

**Reply by felixdittrich92** [reply]
I think anything was broken by building the wheel with pip on your windows machine, because conda comes with precompiled binaries where no wheel is required. :)

**Reply by sergei-voronkov** [reply]
Thanks a lot!
